workflows/workflow.go

```diff
@@ -8,6 +8,7 @@ import (
 type InterfaceWorkflow interface {
 	AddNode(node nodes.InterfaceNode)
 	RunVirt()
```

Comment:
Let's add the `Run()` function to workflow too and make `RunVirt()` private (`runVirt()`)

`Run()` should be defined in Workflow, and `Run()` will call `runVirt()`. The reason for this is if we want to do some logging for every single workflow we run, we don't want every workflow to have to recode all the logging.

Main/WorkflowManager should call `run()` not `runVirt()`

Reply:
I see done - used the abstract class approach previously discussed.